src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -31,93 +96,1203 @@ const ActorInit En_Go_InitVars = {
+                               &D_801333E8);
+
+        this->skelanime.animPlaybackSpeed = 0.1f;
+        this->skelanime.animPlaybackSpeed *= (this->actor.params & 0xF0) == 0x90 ? 0.5f : 1.0f;
+
+        EnGo_SetupAction(this, EnGo_WakeUp);
+        if ((this->actor.params & 0xF0) == 0x90) {
+            func_800800F8(globalCtx, 0x1068, -0x63, &this->actor, 0);
+        }
+    }
+}
+
+void EnGo_WakeUp(EnGo* this, GlobalContext* globalCtx) {
+    f32 float1;
+
+    if (this->skelanime.animPlaybackSpeed != 0.0f) {
+
+        Math_SmoothScaleMaxMinF(&this->skelanime.animPlaybackSpeed,
+                                ((this->actor.params & 0xF0) == 0x90 ? 0.5f : 1.0f) * 0.5f, 0.1f, 1000.0f, 0.1f);
+
+        float1 = this->skelanime.animCurrentFrame;
+        float1 += this->skelanime.animPlaybackSpeed;
+
+        if (float1 <= 12.0f) {
+            return;
+        } else {
+            this->skelanime.animCurrentFrame = 12.0f;
+            this->skelanime.animPlaybackSpeed = 0.0f;
+            if ((this->actor.params & 0xF0) != 0x90) {
+                this->unk_212 = 0x1E;
+                return;
+            }
+        }
+    }
+
+    if (DECR(this->unk_212) == 0) {
+        Audio_PlaySoundGeneral(NA_SE_EN_GOLON_SIT_DOWN, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        EnGo_SetupAction(this, &func_80A405CC);
+    } else if (EnGo_IsCameraModified(this, globalCtx) == 0) {
+        EnGo_SwapInitialFrameAnimFrameCount(this);
+        this->skelanime.animPlaybackSpeed = 0.0f;
+        EnGo_SetupAction(this, func_80A40494);
+    }
+}
+
+void func_80A40494(EnGo* this, GlobalContext* globalCtx) {
+    f32 float1;
+
+    Math_SmoothScaleMaxMinF(&this->skelanime.animPlaybackSpeed,
+                            ((this->actor.params & 0xF0) == 0x90 ? 0.5f : 1.0f) * -0.5f, 0.1f, 1000.0f, 0.1f);
+    float1 = this->skelanime.animCurrentFrame;
+    float1 += this->skelanime.animPlaybackSpeed;
+    if (!(float1 >= 0.0f)) {
+        Audio_PlaySoundGeneral(NA_SE_EN_DODO_M_GND, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        EnGo_SpawnDust(this, 10, 0.4f, 0.1f, 16, 26.0f, 2.0f);
+        EnGo_SwapInitialFrameAnimFrameCount(this);
+        this->skelanime.animPlaybackSpeed = 0.0f;
+        this->skelanime.animCurrentFrame = 0.0f;
+        this->unk_210 = Math_Rand_S16Offset(30, 30);
+        EnGo_SetupAction(this, EnGo_CurledUp);
+    }
+}
+
+void func_80A405CC(EnGo* this, GlobalContext* globalCtx) {
+    f32 frameCount;
+    f32 float1;
+
+    frameCount = SkelAnime_GetFrameCount(&D_06004930);
+    Math_SmoothScaleMaxMinF(&this->skelanime.animPlaybackSpeed, (this->actor.params & 0xF0) == 0x90 ? 0.5f : 1.0f, 0.1f,
+                            1000.0f, 0.1f);
+
+    float1 = this->skelanime.animCurrentFrame;
+    float1 += this->skelanime.animPlaybackSpeed;
+    if (!(float1 < frameCount)) {
```

Comment:
does `float1 >= frameCount` work?

Reply:
It does not unfortunately and I have no idea why. I think there were a small number of places something similar happened too. Not sure if I'm missing something obvious

Reply:
You could also try `frameCount => float1`? Either way, feel free to resolve this comment when you're comfortable with what you've tried

Reply:
rearranging the expression `!(frameCount > float1)` also matches but that's the only combination that also works. Without the not `!`, the mips bc1tl becomes bc1fl and the only way to fix the "false likely" to be the correct "true likely" is to have the negation in there, at least from what I've tried